components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -833,6 +850,29 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
         return consentPageUrl;
     }
 
+    private static String getConsentPageURLWithScopeMetadata(String consentPageURL, Set<String> scopes,
+                                                             String tenantDomain) {
+
+        try {
+            List<String> oidcScopeList = oAuthAdminService.getRegisteredOIDCScope(tenantDomain);
```

Comment:
do we need to filter the oidc scopes if we are passing the consent required scopes to this method.

Reply:
Yes, oidc scopes are still available in the consent required scopes list.